packages/astro/src/core/util.ts

```diff
@@ -161,7 +161,7 @@ export function emoji(char: string, fallback: string) {
  */
 // NOTE: `/@id/` should only be used when the id is fully resolved
 // TODO: Export a helper util from Vite
-export async function resolveIdToUrl(loader: ModuleLoader, id: string) {
+export async function resolveIdToUrl(loader: ModuleLoader, id: string, root?: URL) {
```

Comment:
Ideally the `root` should always be needed, but I find getting this info is tricky for some cases, so I made it optional for now. The other cases should work fine without this fix.